WordPressKit/HTTPRequestBuilder.swift

```diff
@@ -256,9 +256,10 @@ private extension Dictionary where Key == String, Value == Any {
     }
 
     func flatten() -> [URLQueryItem] {
-        reduce(into: []) { result, entry in
-            Self.urlEncode(into: &result, name: entry.key, value: entry.value)
-        }
+        sorted { $0.key < $1.key }
+            .reduce(into: []) { result, entry in
+                Self.urlEncode(into: &result, name: entry.key, value: entry.value)
+            }
```

Comment:
There are unit tests that stubs API requests with certain query string like "foo=1&bar=2". This change here makes sure given the same dictionary `[String: Any]`, the encoded query will always be the same, so that the query lookup in unit tests can pass reliably.

Reply:
This is a tricky one, because it's potentially a breaking change (what happens if I add two `URLQueryItem`s with the same key and different values – I'd expect newer values to overwrite older ones right now – do we still have that guarantee now?)

I'd say the tests should be reworked for this particular case, but WDYT?

Reply:
This `flatten` method is on `[String: Any]` type. So, it's not possible to have two query items that has the same key.

But the overriding problem you mentioned does existing in other cases, for example adding query one by one. Those cases are handled by [the adding query items APIs](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/cb5f349e92110e40a6c9cdc11d3da8cd0f6c71ad/WordPressKit/HTTPRequestBuilder.swift#L62), which has a `override` parameter—the call decides the overriding behaviour. And they are covered by existing unit tests. Are there more tests you think I should add here?

https://github.com/wordpress-mobile/WordPressKit-iOS/blob/cb5f349e92110e40a6c9cdc11d3da8cd0f6c71ad/WordPressKitTests/Utilities/HTTPRequestBuilderTests.swift#L108-L126

Reply:
Sounds like we're covered then, thanks!!